### ISSUE TITLE
Unable to remap numeric keypad keys

### ISSUE THREAD

**Issue by huzzam** [musescore/MuseScore#17080]
### Issue type

UI bug

### Bug description

When I try to remap the numeric keypad keys, MuseScore actually just remaps the regular number keys. For example (since I find the default mapping counterintuitive) I want e.g. '4' on the keypad to be a quarter note. When I remap it, hitting '4' on the keypad doesn't return 'Num4' as it should, but instead '4'. No combination of modifiers gets MuseScore's remapper to recognize Num4...

I've confirmed that other applications that differentiate between keypad and number-row do recognize the difference, such as Pro Tools. Also, Karabiner EventViewer also returns different codes for keypad vs number-row. So I don't think my system has a problem.

This is MuseScore 4.0.2 on two Macs ~ M1 Macbook Air with Ventura 13.3, and i9 iMac with Monterey 12.6.3.

### Steps to reproduce

1. Open Preferences->Shortcuts
2. Find a shortcut to modify, and double click it to open the "Define shortcut" popup
3. Press a numeric keypad key
4. You'll see that instead of e.g. "Num4", the value "4" is defined, which corresponds only to the number 4 on the number-row.
5. Close the dialog, and try to use the new shortcut via the number pad
6. Nothing happens.

### Screenshots/Screen recordings

video on youtube (too big to add here)
https://youtu.be/lLz0cgRSruI

### MuseScore Version

4.0.2

### Regression

Yes, this used to work in Musescore 3.x and now is broken

### Operating system

MacOS 13 and MacOS 12

**Comment by WinstonSung** [comment]
(Got same issue. comment for CC.)

**Comment by Jojo-Schmitz** [comment]
Workaround is to (locale and) edit shortcuts.xml directly

**Comment by WinstonSung** [comment]
> Workaround is to (locate and) edit shortcuts.xml directly

Yeah, I've found it and doing it. But some `<seq>`s need to be updated, e. g., `NumPad` => `Num` (for 3.6 to 4.0.2 migration).

**Comment by gus3000** [comment]
Same issue on Windows 11 with version 4.2.1.240230937.
Editing the `shortcuts.xml` manually still works, though.
(for Windows users reading this, it's located in `C:\Users\gus3000\AppData\Local\MuseScore\MuseScore4`)

**Comment by E5rael** [comment]
I'm running a Flatpak version on Linux Mint (OS: KDE Flatpak runtime, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.2-, revision: github-musescore-musescore-) with the same issue, trying the shortcuts.xml workaround.

I wasn't able to find the shortcut file (probably embedded somewhere inside the flatpak or something), but I was able to export the shortcut file with Musescore itself. So, after modifying the xml file I pressed the "import" button, but the xml file is invisible in the import interface's file explorer, as if it was of a wrong filetype.

**Comment by winstonsung** [comment]
It even failed to recognize NumPad 4 key press as `Num+4` on MuseScore 4.7 Linux AppImage now.

**Comment by winstonsung** [comment]
It appears the problem is from here:

These part of codes removed the `Num`/`NumPad` from the modifiers and caused the regression.

```cpp
modifiers &= ~Qt::KeypadModifier;
```

* (before split repository) https://github.com/musescore/MuseScore/blob/773e6e32/src/framework/shortcuts/shortcutstypes.h#L184-L194
* https://github.com/musescore/muse_framework/blob/93b38f653d4bf1fa7ce6ec57c2ff6374d4130fef/framework/shortcuts/shortcutstypes.h#L109-L119
* https://github.com/musescore/muse_framework/blob/93b38f653d4bf1fa7ce6ec57c2ff6374d4130fef/framework/shortcuts_v2/shortcutstypes.h#L112-L122

**Comment by winstonsung** [comment]
See also:

* https://github.com/musescore/MuseScore/issues/14697

**Comment by winstonsung** [comment]
* https://github.com/musescore/MuseScore/commit/5e4024b2c26144f72b54857f99105a0c84d8461c
* https://github.com/musescore/MuseScore/pull/10746
* https://github.com/musescore/MuseScore/issues/9765
